libs/net/template/Extensions/ReportModelExtensions.cs

```diff
@@ -0,0 +1,54 @@
+using System.Text.Json;
+using TNO.TemplateEngine.Models.Reports;
+
+namespace TNO.TemplateEngine.Extensions;
+
+/// <summary>
+/// ReportModelExtensions static class, provides extension methods for ReportModel objects.
+/// </summary>
+public static class ReportModelExtensions
+{
+    /// <summary>
+    /// Parse the report settings for sections.
+    /// </summary>
+    /// <param name="report"></param>
+    /// <returns></returns>
+    private static Dictionary<string, ReportSectionModel> ParseSections(this JsonDocument settings)
+    {
+        var sectionsElement = settings.RootElement.GetProperty("sections");
+        var sections = new Dictionary<string, ReportSectionModel>();
+        foreach (var sectionElement in sectionsElement.EnumerateArray())
+        {
+            var name = sectionElement.GetProperty("name").GetString() ?? "";
+            var label = sectionElement.GetProperty("label").GetString() ?? "";
+            sections.Add(name, new ReportSectionModel(name, label));
+        }
+        return sections;
+    }
+
+    /// <summary>
+    /// Parse the report settings for sections.
+    /// </summary>
+    /// <param name="report"></param>
+    /// <returns></returns>
+    public static Dictionary<string, ReportSectionModel> ParseSections(this API.Areas.Services.Models.Report.ReportModel report)
+    {
+        var settings = JsonDocument.Parse(JsonSerializer.Serialize(report.Settings));
+        var sections = settings.ParseSections();
+        sections.Add("", new ReportSectionModel("", ""));
```

Comment:
I'm not a fan of how I did this, but it ensures that the main filter with an empty string is also included in the sections.  Having everything as part of the dictionary makes it easy to return everything the same way.